homeassistant/components/pi4ioe5v9xxxx/switch.py

```diff
@@ -62,6 +61,10 @@ def __init__(self, name, pin, invert_logic):
         self._invert_logic = invert_logic
         self._state = False
 
+    def async_added_to_hass(self):
+        """On initializatin time switch off. This is required in case of inversion."""
+        self.turn_off()
```

Comment:
We usually don't actuate changes when adding the entity. Why should we do that?

Reply:
Can't we just set the state of the entity to the state of the device?

Reply:
No, the hardware is write-only if configured as output. What actually happens is the following:
- Hardware I was running is a rpi with I2C connection to these driver chips
- Hardware is started from off to on (power up)
- HA is starting up. All switche-states are off, as they do not remember their state. 
- If the 'inverse' bit for a switch is set, the switch-state 'off' means: hardware should be 'on'
- As the driver chips is in its default startup condition (off), this is not the case. 
- What is more confusing: the state in the inteface shown is the software state, so the state in the interface is 'off', while in hardware it is 'on'.

This should fix that. 

For restarts of HA-software only, saving the state could also fix this, but that does not fix the power-reboot-issue.

Reply:
Yeah, my suggestion is to simply change the internal entity state to match the current hardware state without changing the hardware state.

How do we know that the user wants the hardware to turn off when we restart home assistant?